src/components/Form/Form.js

```diff
@@ -0,0 +1,52 @@
+import { useState } from 'react';
+
+import FormValidator from 'helpers/form-validator';
+// Components
+import Input from 'components/Input/Input';
+
+const Form = (({formData}) => {
+  const defaultFormData = formData;
+  const [data, updatedata] = useState(formData);
+
+  const onChangeHandler = ((event) => {
+    let target = event.target;
+    updatedata({
+      ...data,
+      [target.name]: {...data[target.name], value: target.value, checked: target.checked}
+    });
+  });
+
+  const onFormSubmit = ((event) => {
+    event.preventDefault();
+    if (validateForm()) {
+      alert('Form submitted');
+      updatedata(defaultFormData);
+    }
+  });
+
+  const validateForm = (() => {
+    let formValidator = new FormValidator(),
+      isDataValid = true;
+    Object.keys(data).forEach((key) => {
```

Comment:
much better to create a new state such as formErrors and store your errors there, rather than maintaining one huge state.